packages/dds/tree/src/test/tablePerformanceTestUtilities.ts

```diff
@@ -0,0 +1,190 @@
+ * Defines the schema for a table row.
+ */
+export class Row extends TableSchema.row({
+	schemaFactory,
+	cell: Cell,
+}) {}
+
+/**
+ * Defines the schema for a table, which includes columns and rows.
+ * It uses the previously defined Cell, Column, and Row schemas.
+ */
+export class Table extends TableSchema.table({
+	schemaFactory,
+	cell: Cell,
+	column: Column,
+	row: Row,
+}) {}
+
+/**
+ * Provides a simple table tree initialized with the specified size and cell value.
+ * This helper function creates a table schema, initializes a SharedTree instance,
+ * and populates it with the specified number of rows and columns.
+ * Each cell is initialized with the provided cell value.
+ *
+ * @param tableSize - The number of rows and columns to create in the table.
+ * @param cellValue - The initial value to set in each cell of the table.
+ * @returns A fully initialized table tree definition, including table instance, undo/redo stacks, and a cleanup function.
+ */
+export function createTableTree(tableSize: number, cellValue: string): TableTreeDefinition {
+	const sharedTreeFactory = DefaultTestSharedTreeKind.getFactory();
+	const runtime = new MockFluidDataStoreRuntime({
+		idCompressor: createIdCompressor(),
+		attachState: AttachState.Detached,
+	});
+	const tree = sharedTreeFactory.create(runtime, "tree");
+	const treeView = tree.viewWith(
+		new TreeViewConfiguration({
+			schema: Table,
+			enableSchemaValidation: true,
+		}),
+	);
+
+	treeView.initialize(Table.empty());
+	const undoStack: Revertible[] = [];
+	const redoStack: Revertible[] = [];
+
+	function onDispose(disposed: Revertible): void {
+		const redoIndex = redoStack.indexOf(disposed);
+		if (redoIndex !== -1) {
+			redoStack.splice(redoIndex, 1);
+		} else {
+			const undoIndex = undoStack.indexOf(disposed);
+			if (undoIndex !== -1) {
+				undoStack.splice(undoIndex, 1);
+			}
+		}
+	}
+
+	const unsubscribeFromCommitAppliedEvent = treeView.events.on(
+		"commitApplied",
+		(commit, getRevertible) => {
+			if (getRevertible !== undefined) {
+				const revertible = getRevertible(onDispose);
+				if (commit.kind === CommitKind.Undo) {
+					redoStack.push(revertible);
+				} else {
+					undoStack.push(revertible);
+				}
+			}
+		},
+	);
+	const unsubscribe = (): void => {
+		unsubscribeFromCommitAppliedEvent();
+		for (const revertible of undoStack) {
+			revertible.dispose();
+		}
+		for (const revertible of redoStack) {
+			revertible.dispose();
+		}
+	};
+
+	const table = treeView.root;
+	for (let i = 0; i < tableSize; i++) {
+		const column = new Column({ id: `column-${i}` });
+		table.insertColumn({ index: i, column });
+	}
+	for (let i = 0; i < tableSize; i++) {
+		const row = new Row({ id: `row-${i}`, cells: {} });
+		table.insertRow({ index: i, row });
+	}
+	for (let i = 0; i < tableSize; i++) {
+		for (let j = 0; j < tableSize; j++) {
+			table.setCell({
+				key: {
+					column: `column-${i}`,
+					row: `row-${j}`,
+				},
+				cell: { cellValue },
+			});
+		}
+	}
```

Comment:
Since we no longer seem to depend on the specific format of the ID strings for columns and rows, we should be able to update this to leverage SharedTree's automatic ID generation (which we generally expect users to leverage).

```suggestion
	for (let i = 0; i < tableSize; i++) {
		const column = new Column({});
		table.insertColumn({ index: i, column });
	}
	for (let i = 0; i < tableSize; i++) {
		const row = new Row({ cells: {} });
		table.insertRow({ index: i, row });
	}
	for (const row of table.rows) {
		for (const column of table.columns) {
			table.setCell({
				key: {
					column,
					row,
				},
				cell: { cellValue },
			});
		}
	}
```

Reply:
Applied